src/main/java/io/druid/segment/loading/DataSegmentPuller.java

```diff
@@ -10,10 +13,49 @@
 {
   /**
    * Pull down segment files for the given DataSegment and put them in the given directory.
+   * Deprecated in favor of URI pulling when possible
    *
    * @param segment The segment to pull down files for
-   * @param dir The directory to store the files in
+   * @param dir     The directory to store the files in
+   *
    * @throws SegmentLoadingException if there are any errors
    */
+  @Deprecated
```

Comment:
why did we deprecate this? Do we want the URI to be the standard way to pull down segments? I think there is simplicity in allowing you to just pass a segment and not having to worry about where that segment is stored.